src/runtime/common/serialization.cpp

```diff
@@ -0,0 +1,178 @@
+
+#include "hipSYCL/common/serialization.hpp"
+#include "hipSYCL/sycl/access.hpp"
+#include "hipSYCL/sycl/item.hpp"
+#include "hipSYCL/sycl/range.hpp"
+#include "hipSYCL/runtime/backend.hpp"
+#include "hipSYCL/runtime/operations.hpp"
+#include "hipSYCL/runtime/dag.hpp"
+#include "hipSYCL/runtime/device_id.hpp"
+
+#include <iostream>
+#include <map>
+
+
+namespace hipsycl::sycl::access{
+std::ostream& operator<<(std::ostream& out, const mode value){
+    static std::map<mode, std::string> strings;
+    if (strings.size() == 0){
+        strings[mode::read] = "R";
+        strings[mode::write] = "W";
+        strings[mode::atomic] = "atomic";
+        strings[mode::read_write] = "RW";
+        strings[mode::discard_write] = "Discard W";
+        strings[mode::discard_read_write] = "Discard RW";
+    }
+    return out << strings[value];
+}
+
+std::ostream& operator<<(std::ostream& out, const target value){
+    static std::map<target, std::string> strings;
+    if (strings.size() == 0){
+#define INSERT_ELEMENT(p) strings[p] = #p
+        INSERT_ELEMENT(target::image);
+        INSERT_ELEMENT(target::image);
+        INSERT_ELEMENT(target::host_image);
+        INSERT_ELEMENT(target::host_buffer);
+        INSERT_ELEMENT(target::image_array);
+        INSERT_ELEMENT(target::global_buffer);
+        INSERT_ELEMENT(target::constant_buffer);
+#undef INSERT_ELEMENT
+    }
+    return out << strings[value];
```

Comment:
formatting... Also, why don't you use just the same approach here as for `mode`?